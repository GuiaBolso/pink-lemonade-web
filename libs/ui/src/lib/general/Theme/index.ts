import { customThemeProps as TCustomThemeProps } from './Theme';

export { default as Theme } from './Theme';
export type customThemeProps = TCustomThemeProps;
