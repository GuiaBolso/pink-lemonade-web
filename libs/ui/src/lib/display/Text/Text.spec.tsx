import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Theme } from '../../general';
import { Text } from '.';

describe('Text Component', () => {
  const TextDisplay01 = () => (
    <Theme>
      <Text variant="display-01">Componente de texto</Text>
    </Theme>
  );

  const TextDisplay01Connect = () => (
    <Theme theme="connect">
      <Text variant="display-01">Componente de texto</Text>
    </Theme>
  );

  const Texth1 = () => (
    <Theme>
      <Text component="h1">Componente h1</Text>
    </Theme>
  );

  it('renders correctly', () => {
    const tree = renderer.create(<TextDisplay01 />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders connect theme correctly', () => {
    const tree = renderer.create(<TextDisplay01Connect />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Displays the correct text', () => {
    const { getByTestId } = render(<TextDisplay01 />);

    expect(getByTestId('text')).toHaveTextContent(/^Componente de texto$/);
  });

  it('Displays h1 component', () => {
    const { getByTestId } = render(<Texth1 />);

    expect(getByTestId('text')).toContainHTML('h1');
  });
});
