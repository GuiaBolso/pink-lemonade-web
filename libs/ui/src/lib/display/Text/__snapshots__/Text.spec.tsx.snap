// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component renders connect theme correctly 1`] = `
<span
  className="css-95avy4"
  data-testid="text"
>
  Componente de texto
</span>
`;

exports[`Text Component renders correctly 1`] = `
<span
  className="css-95avy4"
  data-testid="text"
>
  Componente de texto
</span>
`;
